### ISSUE TITLE
How to change mujoco model parameters to pytorch tensor?

### ISSUE THREAD

**Issue by ZJP-c4cld** [openai/gym#2810]
I want to construct a neural network in pytorch which outputs appropriate values of mujoco model parameters.  But this requires that mujoco model parameters are converted into pytorch tensors and mujoco environments run well with these converted model parameters. I don't know how to achieve this goal. Anyone know how to do that?

**Comment by pseudo-rnd-thoughts** [comment]
As most gym environments return numpy arrays then `torch.Tensor(obs)` should work most of the time

While this tutorial is not for mujoco, it should provide you with most of the details you are interested in  https://pytorch.org/tutorials/intermediate/reinforcement_q_learning.html

**Comment by ZJP-c4cld** [comment]
@pseudo-rnd-thoughts Thank you for your prompt reply and selfless help ^_^
Most gym environments are written only by python, so converting model parameters into pytorch tensors and computing gradient of reward w.r.t model parameters are easy things. 
In contrast, it is a different story when it comes to Mujoco, because Mujoco is activated by a simulator written by C++. Computing gradient of reward w.r.t model parameters is a difficult issue. Do you have any advice?

**Comment by pseudo-rnd-thoughts** [comment]
Im not sure that I understand what you are saying. 
For deep reinforcement learning algorithms, we compute the gradient using the network parameter w.r.t the training loss. 
So we do not find the gradient with respect to the environment implementation and it doesn't matter if the environment is written in python or c++. 
The mujoco environments return a numpy array like cartpole for example so you should be able to use the observation in the same way.

**Comment by ZJP-c4cld** [comment]
@pseudo-rnd-thoughts Thank you for your joining in the discussion. ^_^
